beacon-chain/core/altair/epoch_precompute.go

```diff
@@ -177,27 +173,28 @@ func ProcessEpochParticipation(
 // ProcessRewardsAndPenaltiesPrecompute processes the rewards and penalties of individual validator.
 // This is an optimized version by passing in precomputed validator attesting records and and total epoch balances.
 func ProcessRewardsAndPenaltiesPrecompute(
-	state state.BeaconStateAltair,
+	beaconState state.BeaconStateAltair,
 	bal *precompute.Balance,
 	vals []*precompute.Validator,
 ) (state.BeaconStateAltair, error) {
 	// Don't process rewards and penalties in genesis epoch.
-	if core.CurrentEpoch(state) == 0 {
-		return state, nil
+	cfg := params.BeaconConfig()
+	if core.CurrentEpoch(beaconState) == cfg.GenesisEpoch {
+		return beaconState, nil
 	}
 
-	numOfVals := state.NumValidators()
+	numOfVals := beaconState.NumValidators()
 	// Guard against an out-of-bounds using validator balance precompute.
-	if len(vals) != numOfVals || len(vals) != state.BalancesLength() {
-		return state, errors.New("validator registries not the same length as state's validator registries")
+	if len(vals) != numOfVals || len(vals) != beaconState.BalancesLength() {
+		return beaconState, errors.New("validator registries not the same length as beaconState's validator registries")
```

Comment:
```suggestion
		return beaconState, errors.New("validator registries not the same length as state's validator registries")
```

Unwanted refactoring side-effect.